src/nested.jl

```diff
@@ -0,0 +1,54 @@
+# ------------------------------------------------------------------
+# Licensed under the ISC License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+# helper function to extract raw data
+# from uniform scaling objects
+raw(a::UniformScaling) = a.λ
+raw(a) = a
+
+"""
+    NestedVariogram(cs, γs)
+
+A nested variogram model `γ = c₁γ₁ + c₂γ₂ + ⋯ + cₙγₙ` with
+coefficients `cs = (c₁, c₂, ..., cₙ)` and variogram models
+`γs = (γ₁, γ₂, ..., γₙ)`.
+"""
+struct NestedVariogram{N} <: Variogram{Number,Metric}
+  cs::NTuple{N,Any}
+  γs::NTuple{N,Variogram}
+end
+(g::NestedVariogram)(h) = raw(sum(c*γ(h) for (c, γ) in zip(g.cs, g.γs)))
+(g::NestedVariogram)(x, y) = raw(sum(c*γ(x,y) for (c, γ) in zip(g.cs, g.γs)))
```

Comment:
Only error I got were in these cases with mixed scalars
I think it's more an user error than anything else. We can either ignore it or change to `c*I*γ(h)` (will assume that a scalar will add only to the direct variograms of the diagonal)

```julia
z = [5.1 0; 0 2] * GaussianVariogram(range=1.0) + 10 * SphericalVariogram(range=2.0)
z([0,0],[19,25]) # or z(10)

MethodError: no method matching +(::Array{Float64,2}, ::Float64)
```

Reply:
Nice catch. Maybe we shouldn't support mixed coefficients? The issue seems to point out that users should use `10I` as the coefficient in your example, right?

Regardless of this decision, I wonder makes more sense in general: multiply the diagonal or all entries of the matrices in the various functions. In other words, should we use `c*I*γ(h)` or `c*γ(h)` in the various functions, including `sill` and `nugget`. I don't have much experience with co-estimation/co-simulation, but it would be nice to choose a sensible design. We can always fix things in future PRs too. What do you think?

Reply:
We can start with univariate case and then when we are happy with the API, extend it to multivariate. Ideally, if the design is prepared for both cases, the better.

Reply:
Agreed. Let's keep `c*γ(h)`. Don't know the best solution to deal with mixed cases, better throw an error when mixing matrices with scalars right now